client/reader/controller.js

```diff
@@ -55,6 +57,23 @@ function trackScrollPage( path, title, category, readerView, pageNum ) {
 	} );
 }
 
+function onRouteChange( newContext, oldContext ) {
+	if ( ! oldContext ) {
+		return;
+	}
+
+	const fullPostViewPrefix = '/read/post/';
+
+	if ( startsWith( oldContext.path, fullPostViewPrefix ) &&
+		! startsWith( newContext.path, fullPostViewPrefix ) &&
+		__fullPostInstance ) {
+		__fullPostInstance.setState( { isVisible: false } );
+		__fullPostInstance = false;
+	}
+}
+
+routeChange( onRouteChange );
```

Comment:
Good point, I'll change it.

Random aside, but how do arrow funcs appear in the debugger? One nice thing about using a named function is that the call stack has useful information.